### DISCUSSION TITLE
Seeking guidance on how to maintain project files between two computers

### DISCUSSION THREAD

**Discussion by Ubuntufoo** [community/community#46018] [New to GitHub]
I have a github repo set up and fully functional. My concern is how to utilize two PC's without corrupting files. What is the ideal way to do this?
My standard environments are identical: windows 10, same ver of Python/Django. IDE is VSC with GitHub Pull Requests and Issues extension. My plan is to create a branch to be safe, pull it to my 2nd device and create a new virtual environment on it to continue working.

What are some considerations before doing this?

Thank you kindly

**Comment by chris-c-thomas** [comment]
Well you see this is an interesting question... as it's literally what Git was invented to do. I don't want anything to come off like I'm making any assumptions about where you're at with your skills with it so I might end up being captain obvious about a few things.

[Wikipedia](https://en.wikipedia.org/wiki/Git) page because I just read through some parts myself again for fun and I prefer it to the git scm stuff.

Git makes a lot more sense sometimes when you have a lot of people working on the same project. The goal was to create a source control with history of changes made throughout the code structure but also allow many people to work on it at the same time without overriding one another constantly. So just think of yourself as two people in this scenario.

You'll have your `main` branch and that can be considered your production environment. Maybe create a `develop` branch and that can be your staging type area that new features get pulled into when they are ready. Many workflows exist around the topic.

If you planned to work on two separate features, one on each computer, then I'd personally create separate branches based off of `develop` for each computer to work on. When one is done with a feature, a pull request will be opened, and can be merged into `develop`. You can merge the second one in afterwards or update it to contain the new version of `develop` if you end up needing those changes. If both branches touched the same file, you'll get a merge conflict, and you'll have to decide what changes to keep and what to discard where each discrepancy exists.

If you just plan to switch back and forth between computers as you work on the same thing then just take your changes from your current computer, push them up to whatever branch you're working on up to GitHub.com, and pull those changes back down on your second computer. I'd still personally use separate branches for new features, bug fixes, etc and merge them into a `develop` type branch when the features or fixes are done. That branch can then be thoroughly tested and if all is well merged into `main` for production use.

**Comment by Steveplays28** [comment]
VSCode has a feature called Cloud Changes, which allows you to synchronize git changes between computers without having to commit.

Hope I interpreted your question right, if not, ignore this answer.

**Reply by chris-c-thomas** [reply]
Oh interesting! I wasn't familiar with that one. I can't find Cloud Changes in their docs, is it newer?

So no commits to the actual repo but any changes I make in a file will reflect if I open the workspace elsewhere?

**Reply by Steveplays28** [reply]
Yeah it's only a few months old and docs are nonexistant. Just a blog post *somewhere*.

It makes no commits in the actual repo indeed. It also checks what branch you're on when syncing cloud changes, so you can't accidentally pull cloud changes into the wrong branch.
It's also not automatic, but you could probably use an extension or something to run the sync cloud changes command on save or something similar.

**Comment by paule32** [comment]
Hello,
First, a good practise is "to have" daily or periodly (all two hours) a Back-Up of the files of each Computer on local storage or local servers.
Not all Developer work like the others. So the speed of development is very dynamically.
If you have made a Back-Up, a good practise is, to use Software like "git". Or use the WebSoftware Service github - it based on the software application called "git".
If you dont want to publish your codes, you can install git locally.
But then, you have no comfort like there on github.
Visual Studio 2020 comes with Git Module, so you can store locally, and remote your Back-Ups.
You need a Time planer, on which your Projects will be commit, and push to the servers.
On remote servers, this can take some seconds, or minutes, because git verifery very much things and check against newer Version of your BackUp files. It could be a hole, but please take attention to files that you save in the past local, and fetch it with new ones from the remote server.
So,  I would give you a advise: make a Plan, which developer should work on which source codes.
So, you can make a shedule Plan, that developer A take the job of implement the Menu-System of a Application, where the developer B take the job to implement the Text-Editor. This was examples.
A good thing is an other Plan, where you draw back a time line for each developer, and each task.
So, if you set a deadline where the release server compiles the stuff all together, what the developers collect on source code,
You can decide, if the work of the developer that came to late is involved.
Please sorry for my English. It is very bad. But I would give some hints on make things better.
Maybe someone helps this.
Thanks for reading, paule32

**Comment by Ubuntufoo** [comment]
I appreciate all the answers. I'm not on a team of developers -  just a student learning everything from scratch. Still wrapping my head around file dependencies, virtual environments, etc... Likely things that seasoned programmers take for granted. Amazingly, all the sources I've found have a complex way of explaining how to simply grab files from a repo in a basic cloud storage capacity.

I went back to my source control tab, clicked on "pull", and I have my files on my second device now. Cheers.

**Reply by chris-c-thomas** [reply]
Glad you have a solution working for yourself now!
